vision_datasets/kv_pair/manifest.py

```diff
@@ -0,0 +1,128 @@
+            raise ValueError(f'enum is only allowed for string, number, integer types')
+        if self.type == 'array' and not self.items:
+            raise ValueError(f'items must be provided for array type')
+        elif self.type == 'object' and not self.properties:
+            raise ValueError(f'properties must be provided for object type')
+
+
+class KVPairSchema:
+    def __init__(self, name: str, field_schema_dict: dict, description: str = None) -> None:
+        self.name = name
+        self.description = description
+        self.field_schema = {k: KVPairFieldSchema(**v) for k, v in field_schema_dict.items()}
+    
+    def __eq__(self, other) -> bool:
+        return self.name == other.name and self.field_schema == other.field_schema and self.description == other.description
+
+
+class KVPairLabelManifest(MultiImageLabelManifest):
+    """
+    {
+        "key_value_pairs": {"key1": "value1", ...},
+        "text_input": "optional text input for this annotation"
+    }
+    """
+    LABEL_KEY = 'key_value_pairs'
+    INPUT_KEY = 'text_input'
+    
+    @property
+    def key_value_pairs(self) -> dict:
+        return self.label_data[self.LABEL_KEY]
+
+    @property
+    def text_input(self) -> Optional[dict]:
+        return self.label_data.get(self.INPUT_KEY, None)
+
+    def _read_label_data(self):
+        raise NotImplementedError
+    
+    def _check_label(self, label_data):
+        if not isinstance(label_data, dict) or self.LABEL_KEY not in label_data:
+            raise ValueError
+    
+    @classmethod
+    def check_schema_match(cls, key_value_pairs: dict, schema: KVPairSchema):
+        for key, field_schema in schema.field_schema.items():
+            if key not in key_value_pairs:
+                raise ValueError(f'{key} not found')
+            KVPairLabelManifest.check_field_schema_match(key_value_pairs[key], field_schema)
+    
+    @classmethod
+    def check_field_schema_match(cls, value, field_schema: KVPairFieldSchema):
+        if not isinstance(value, KVPairFieldSchema.TYPE_NAME_TO_PYTHON_TYPE[field_schema.type]):
+            raise ValueError(f'{value} is not of type {field_schema.type}')
+        if field_schema.enum:
+            if value not in field_schema.enum:
+                raise ValueError(f'{value} not in enum {field_schema.enum}')
+        if field_schema.type == 'boundingBox':
+            if len(value) != 4:
+                raise ValueError(f'boundingBox must have 4 elements')
+        elif field_schema.type == 'array':
+            for array_item in value:
+                cls.check_field_schema_match(array_item, field_schema.items)
+        elif field_schema.type == 'object':
+            for k, v in value.items():
+                if k not in field_schema.properties:
+                    raise ValueError(f'{k} not found in schema')
+                cls.check_field_schema_match(v, field_schema.properties[k])
+
+
+class KVPairDatasetManifest(MultiImageDatasetManifest):
```

Comment:
Looking at this again, I wonder if it would be better to change everything currently named `KVPair*` to `KeyValuePair*`. The name isn't that long, is much more readable, and is consistent with the other namings (e.g., `MultiTask*`.